selfservice/hook/web_hook.go

```diff
@@ -387,6 +400,18 @@ func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
 }
 
 func parseWebhookResponse(resp *http.Response, id *identity.Identity) (err error) {
+	defer func() {
+		if err != nil {
+			err = herodot.DefaultError{
+				CodeField:     http.StatusBadGateway,
+				StatusField:   http.StatusText(http.StatusBadGateway),
+				GRPCCodeField: grpccodes.Aborted,
+				ReasonField:   "A third-party upstream service responded improperly. Please try again later.",
+				ErrorField:    err.Error(),
+			}.WithWrap(errors.WithStack(err))
+		}
+	}()
```

Comment:
This is the issue for the failing tests. 

Do we actually always want to return 502 here? If the webhook target responded 400, we interpret that as "a validation error occurred". Reason being, that we can parse the response and show validation errors in the form, for example. 
![image](https://user-images.githubusercontent.com/5540255/231771554-a8fbaff5-bed8-4881-8707-31503bbb0b96.png)

Reply:
Yep, makes sense. Fixed. Thanks for your help!